src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1250,24 +1209,24 @@ public static void setRoleManager(IRoleManager roleManager)
 
     public static int getPermissionsValidity()
     {
-        return conf.permissions_validity_in_ms;
+        return conf.permissions_validity.toMillisecondsAsInt();
     }
 
     public static void setPermissionsValidity(int timeout)
     {
-        conf.permissions_validity_in_ms = timeout;
+        conf.permissions_validity = CassandraDuration.inMilliseconds(timeout);
     }
 
     public static int getPermissionsUpdateInterval()
     {
-        return conf.permissions_update_interval_in_ms == -1
-             ? conf.permissions_validity_in_ms
-             : conf.permissions_update_interval_in_ms;
+        return conf.permissions_update_interval.toMilliseconds() == 0
```

Comment:
This is safe because `MillisDurationConverterCustom` apprehends the `-1` and changes it to zero? I'd perhaps throw a comment in that converter about that. (Also thinking a bit more generally about whether we should add a "previous default" field to the `Replaces` annotation to avoid this being a one-off hack.)

Reply:
My line of thinking is that now when we do this cleaning, after this patch, I expect the framework to be used only for new names but not more similar changes like that. In that case I don't see a reason to overwhelm more the annotation.